### PR TITLE
Improve the reliability of `mvn verify`

### PR DESCRIPTION
**Pull request by bjchambers**
These commits make the following changes to improve reliability of `mvn verify` and interaction with IDEs:

1. Add some casts that were causing complaints.
2. Use separate modules for Java 8 tests (subverted from @kennknowles) and examples (following same pattern).
3. Stop adding the JAR to the surefire tests, since we have the classes anyways.

### CODE REVIEW

**Comment by bjchambers** [comment]
R: @kennknowles

File: java8examples/src/main/java/com/google/cloud/dataflow/examples/complete/game/GameStats.java [hunk 1]
```diff
@@ -76,8 +75,8 @@
  * subsequent fixed windowing. (This could be used to tell us what games are giving us greater
  * user retention).
  *
- * <p> Run {@link injector.Injector} to generate pubsub data for this pipeline.  The Injector
- * documentation provides more detail.
+ * <p> Run {@literal com.google.cloud.dataflow.examples.complete.game.injector.Injector} to generate
```
Comment (kennknowles) [inline java8examples/src/main/java/com/google/cloud/dataflow/examples/complete/game/GameStats.java]: `@code`. I'll fix up when merging.

**Comment by kennknowles** [comment]
LGTM. I'll wait for CI and then merge.

**Comment by davorbonaci** [comment]
We need to fix this up a little -- this is polluting the top-level directory structure.

I think `examples/java/...` and `examples/java8/...` would have been a better choice.

**Pull request merged**